src/storage/storage.ts

```diff
@@ -78,6 +84,27 @@ export class Storage implements IStorage {
     this.version = version;
   }
 
+  /**
+   * Set the tx signing function
+   * @param {EcdsaTxSign} txSign The signing function
+   */
+  setTxSign(txSign: EcdsaTxSign): void {
```

Comment:
Having the storage hold this method makes more sense since this method will be responsible for fetching the private key, identifying which inputs to sign (and the address indexes to use) then signing.
The storage is expected to hold the private keys and addresses information.
In the future the wallets can just override this method on the storage and all internals of the wallet-lib will work.

It also makes the wallet-lib implementation simpler since some internal utilities use the storage instance and not the wallet facade instance, we would need to change them and take care to always pass the wallet facade anywhere we would need to sign anything.

Reply:
If you think we should, we can also have a facade method to set this on the storage.
What do you think?

Reply:
I agree that it's good to have it on storage because of your explanation but I think it's good also to have it on the facade. 

I think that any external client that wants to use this method shouldn't have to call the storage. So I like your suggestion, to create a method in the facade that calls this storage method.

Reply:
Done